### PR TITLE
Add: bundled domain notice to the checkout page

### PR DESCRIPTION
**Pull request by taggon**
#### Changes proposed in this Pull Request

* Shows information about bundled domains when a user purchases dotcom plans.

Requested in pau2Xa-18u-p2#comment-4655

#### Testing instructions

* Add a dotcom paid plan to the cart.
* If the plan bundles a domain, a message about bundled domains should show up in the checkout page as follows:
  Checkout_—_WordPress_com

_Note: Currently every dotcom plan provides a bundled domain._

### CODE REVIEW

**Comment by akirk** [comment]
@taggon would be great if we could use String Freeze when the copy is finished. I added a copy review request label because I think the text could be improved before translation.

**Comment by michelleweber** [comment]
For copy, can we include a link?

Also, just to confirm: it's odd to say "your year-long plan gives you features for two years." Is this only meant to apply to two-year plans, and should there be two versions of this copy, like:

Purchasing a one-year subscription to a WordPress.com plan gives you one year of access to your plan's features and one year of a custom domain name. To select your custom domain, follow these registration instructions [LINK].

Purchasing a two-year subscription to a WordPress.com plan gives you two years of access to your plan's features and one-year of a custom domain name. To select your custom domain, follow these registration instructions [LINK]. After the first year, you'll continue to have access to your WordPress.com plan features but will need to renew the domain name.

**Comment by taggon** [comment]
> For copy, can we include a link?

@michelleweber Sure. It is also possible to show different copies based on the selected plan years.

**Comment by taggon** [comment]
Now the copy changes based on the length of the selected subscription as per @michelleweber's suggestion. The registration instructions link in the copy heads for [Register a New Domain](https://wordpress.com/support/domains/register-domain/) page.

**One-year subscription**
Checkout_—_WordPress_com

**Two-year subscription**
Checkout_—_WordPress_com

@semihakocer @michelleweber Any thoughts?

**Comment by semihakocer** [comment]
Looks great! Thanks Taegon.

**Comment by taggon** [comment]
We're going to deploy this as soon as it gets approved even if the strings are not translated as per @semihakocer's request. For non-English users, the additional copy shows up only when the translation is ready. cc @akirk

**Comment by DavidRothstein** [comment]
It seems wrong to me that we're giving people instructions on how to "select your custom domain" when they've already done so and the domain is already in the cart.... Shouldn't that sentence (or perhaps even the entire bullet point) be removed in that scenario?

domain-in-cart

**Comment by semihakocer** [comment]
Rethinking this, I think @DavidRothstein is right. The link could create extra confusion. Let's remove only the link and the remaining text remains to give extra information. cc: @taggon @michelleweber

**Comment by taggon** [comment]
Thanks @DavidRothstein for the review! Now the registration link is hidden when the cart includes a domain registration.

Checkout_—_WordPress_com

**Comment by DavidRothstein** [comment]
Hm, I actually think removing the link but leaving the text is more confusing.  We're still telling them they need to select a domain and follow instructions, but now no longer even telling them what instructions to follow....

Why not just remove the "To select your custom domain, follow the registration instructions" sentence entirely in this scenario?  They already selected a domain, so they don't need it.  And any further instructions they need to set it up are given to them later, after they purchase.

**Comment by semihakocer** [comment]
I've re-tested the sign-up flow today to make sure customers receive enough info about domain onboarding. That's ok to remove the last sentence per @DavidRothstein's recommendation. cc: @taggon

**Comment by taggon** [comment]
The copies have been updated. Hope everyone here is happy with the changes. The followings show all scenarios of this PR. cc @semihakocer @DavidRothstein 

Checkout_—_WordPress_com
Checkout_—_WordPress_com-2
Checkout_—_WordPress_com-3
Checkout_—_WordPress_com-4

**For non-English users(when the translation is not ready)**
체크아웃_—_WordPress_com

File: client/my-sites/checkout/checkout/bundled-domain-notice.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	// The plan should bundle a free domain
+	if ( ! isNextDomainFree( cart ) ) {
+		return null;
+	}
+
+	// Hide non-translated text for non-English users.
+	// TODO: the following lines of code should be removed once all translations are ready.
+	if (
+		! translationExists(
+			'Purchasing a one-year subscription to a WordPress.com plan gives you one year of access to your plan’s features and one year of a custom domain name.'
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout/bundled-domain-notice.jsx:49]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/checkout/checkout/bundled-domain-notice.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	);
+
+	const oneYearCopy = translate(
+		'Purchasing a one-year subscription to a WordPress.com plan gives you one year of access to your plan’s features and one year of a custom domain name.',
+		{
+			context: 'checkout',
+		}
+	);
+	const twoYearCopy = translate(
+		'Purchasing a two-year subscription to a WordPress.com plan gives you two years of access to your plan’s features and one year of a custom domain name.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout/bundled-domain-notice.jsx:66]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/checkout/checkout/bundled-domain-notice.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		{
+			context: 'checkout',
+		}
+	);
+	const afterFirstYear = translate(
+		'After the first year, you’ll continue to have access to your WordPress.com plan features but will need to renew the domain name.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout/bundled-domain-notice.jsx:72]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/checkout/checkout/bundled-domain-notice.jsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		{
+			comment: 'After the first year of the bundled domain...',
+			context: 'checkout',
+		}
+	);
+	const registrationLink = translate(
+		'To select your custom domain, follow {{domainRegistrationLink}}the registration instructions{{/domainRegistrationLink}}.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout/bundled-domain-notice.jsx:79]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

**Comment by semihakocer** [comment]
Looks all good to me. Thanks for the detailed screenshots and explanation @taggon

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3502887).

**Comment by akirk** [comment]
^ Sorry, this is not yet translated (as you can see at the link). We had a bug in determining the translated-ness, I have updated the message.

**Pull request merged**